src/evaluation.py

```diff
@@ -517,6 +551,6 @@ def evaluate_concatenated_response(concatenated_response, evaluation_config):
         )
         current_score += delta
 
-    evaluation_config.conditionally_print_explanation()
+    evaluation_config.conditionally_print_explanation(file_id)
```

Comment:
let's call a separate function instead of coupling the logic with the dedicated small functions